controllers/mdutil/util.go

```diff
@@ -563,7 +563,7 @@ func IsSaturated(deployment *clusterv1.MachineDeployment, ms *clusterv1.MachineS
 		return false
 	}
 	desiredString := ms.Annotations[clusterv1.DesiredReplicasAnnotation]
-	desired, err := strconv.Atoi(desiredString)
+	desired, err := strconv.ParseInt(desiredString, 10, 64)
```

Comment:
```suggestion
	desired, err := strconv.ParseInt(desiredString, 10, strconv.IntSize)
```

Though it looks like we are casting below to an int32, maybe it should actually be 32